src/Input/Input.tsx

```diff
@@ -0,0 +1,61 @@
+import { Person } from '../types/Person';
+
+type Props = {
+  appliedQuery: string,
+  showPeople?: boolean,
+  selectedPerson: Person | null,
+  setSelectedPerson: (person: Person | null) => void,
+  handleQueryChange: (event: React.ChangeEvent<HTMLInputElement>) => void,
+  setAppliedQuery?: (string: string) => void,
+  setShowPeople?: (show: boolean) => void,
+  query?: string,
+  setQuery: (string: string) => void,
+};
+
+export const Input: React.FC<Props> = ({
+  appliedQuery,
+  showPeople = false,
+  selectedPerson = null,
+  handleQueryChange,
+  setShowPeople = () => { },
+  setAppliedQuery = () => { },
+  setSelectedPerson = () => { },
+  query,
+  setQuery = () => {},
+}) => {
+  const handleSelectedPerson = () => {
+    if (selectedPerson === null) {
+      return query;
+    }
+
+    return selectedPerson.name;
+  };
+
+  return (
+    <div className="dropdown-trigger  control has-icons-right">
+      <input
+        type="text"
+        placeholder="Enter a part of the name"
+        className="input"
+        value={handleSelectedPerson()}
```

Comment:
Check your demo. Because sorting is not working) 
Also try not to use ( ) in handleSelectedPerson( ).